commons/src/main/java/io/aiven/kafka/connect/common/config/CommonConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.config;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+public abstract class CommonConfig extends AbstractConfig {
```

Comment:
Do we need CommonConfig, while we have AivenCommonConfig, or can we combine those ?

Reply:
CommonConfig is required as AivenCommoonConfig is really sink specific and can not be re-used due to private validation methods for sink specific configuration.

Reply:
Yea, then in that case, can we just combine AivenCommonConfig and SinkConfig ? Now we have several of those configs, which is confusing.

Reply:
AivenCommonConfig is deprecated so it will be removed in the future, but we have to update all connectors before that occurs. It is marked as deprecated so JavaDocs etc and IDE's will highlight it as such when in use.